nuntium/user_section/views.py

```diff
@@ -333,10 +338,33 @@ class ConfirmationTemplateUpdateView(UpdateTemplateWithWriteitBase):
     model = ConfirmationTemplate
 
 
-class MessagesPerWriteItInstance(LoginRequiredMixin, ListView):
+class MessagesPerWriteItInstance(LoginRequiredMixin, ListView, FormMixin):
     model = Message
     template_name = 'nuntium/profiles/messages_per_instance.html'
 
+    def post(self, request, *args, **kwargs):
+
+        contact_ids = request.POST.getlist('contact_ids')
+        
+        outboundmessages = OutboundMessage.objects.filter(message_id = request.POST['message'], contact_id__in = contact_ids)
+ 
+        messages = ''
+
+        for message in outboundmessages:
+            message.status = 'ready'
+            outboundrecords = message.outboundmessagepluginrecord_set.filter(plugin__name = 'mail-channel')
+
+            for outboundrecord in outboundrecords:
+                outboundrecord.try_again = True
+                outboundrecord.save()
+ 
+            message.save()
+            messages += str(message) + '<br/>'
+
+        view_messages.info(request, 'Message resent: ' + messages)
+        
+        return HttpResponseRedirect(reverse('messages_per_writeitinstance', subdomain=self.request.subdomain))
```

Comment:
we have a bit of a unit test, but we also need a test of this view which makes use of the idea.

Doesnt need to be a GUI test. Just something like

client.post(reverse(...), data={contact_ids: [...]} and then checking that the message is in ready state and record is in try-again state.